tests/test_multistore_file.py

```diff
@@ -0,0 +1,44 @@
+"""Unit tests for oauth2client.multistore_file."""
+
+import errno
+import os
+import tempfile
+import unittest
+
+from oauth2client import multistore_file
+
+
+class MultistoreFileTests(unittest.TestCase):
+
+    def test_lock_file_raises_ioerror(self):
+
+        filehandle, filename = tempfile.mkstemp()
+        os.close(filehandle)
+
+        try:
+            for error_code in (errno.EDEADLK, errno.ENOSYS, errno.ENOLCK):
+
+                class MockLockedFile(object):
+
+                    def __init__(self, filename_str, error_code):
+                        self.filename_str = filename_str
+                        self.error_code = error_code
+
+                    def open_and_lock(self):
+                        raise IOError(self.error_code, '')
+
+                    def is_locked(self):
+                        return False
+
+                    def filename(self):
+                        return self.filename_str
+
+                multistore = multistore_file._MultiStore(filename)
+                multistore._file = MockLockedFile(filename, error_code)
+
```

Comment:
After pulling the class out of this function, drop all the blank lines in this function except this one? (Or push back if your aesthetic sense guides you otherwise.)